weblab/fitting/views.py

```diff
@@ -218,24 +218,51 @@ class FittingResultCreateView(LoginRequiredMixin, PermissionRequiredMixin, UserF
 
     def get_initial(self):
         initial = super().get_initial()
-        if 'model' in self.request.GET:
-            initial['model'] = get_object_or_404(ModelEntity, pk=self.request.GET['model'])
-        elif 'protocol' in self.request.GET:
-            initial['protocol'] = get_object_or_404(ProtocolEntity, pk=self.request.GET['protocol'])
-        elif 'fittingspec' in self.request.GET:
-            initial['fittingspec'] = get_object_or_404(FittingSpec, pk=self.request.GET['fittingspec'])
-        elif 'dataset' in self.request.GET:
-            initial['dataset'] = get_object_or_404(Dataset, pk=self.request.GET['dataset'])
+        model_id = self.request.GET.get('model')
+        model_version_id = self.request.GET.get('model_version')
+        protocol_id = self.request.GET.get('protocol')
+        protocol_version_id = self.request.GET.get('protocol_version')
+        fittingspec_id = self.request.GET.get('fittingspec')
+        fittingspec_version_id = self.request.GET.get('fittingspec_version')
+        dataset_id = self.request.GET.get('dataset')
+
+        if model_version_id:
+            initial['model_version'] = get_object_or_404(
+                CachedModelVersion.objects.visible_to_user(self.request.user),
+                pk=model_version_id)
+            initial['model'] = initial['model_version'].model
+        elif model_id:
+            initial['model'] = get_object_or_404(
+                ModelEntity.objects.visible_to_user(self.request.user),
+                pk=model_id)
+
+        elif protocol_version_id:
```

Comment:
I'd be inclined to make this an `if` not an `elif`. We might not supply both model & protocol in normal usage, but why prevent future use when we don't need to? Similarly for `fittingspec_version_id` and `dataset_id`.